src/app.rs

```diff
@@ -186,7 +193,17 @@ impl MatchEvent for App {
                 self.update_login_visibility();
                 self.ui.redraw(cx);
             }
-
+            if let Some(popup_action) = action.downcast_ref() {
+                match popup_action {
+                    PopupNotificationAction::Open => {
+                        self.ui.popup_notification(id!(popup)).open(cx);
+                    }
+                    PopupNotificationAction::Close => {
+                        self.ui.popup_notification(id!(popup)).close(cx);
+                    }
+                    PopupNotificationAction::None => {}
+                }
+            }
```

Comment:
nit: you can combine the `match` and `if` statement:
```suggestion
            match action.downcast_ref() {
                Some(PopupNotificationAction::Open) => {
                    self.ui.popup_notification(id!(popup)).open(cx);
                }
                Some(PopupNotificationAction::Close) => {
                    self.ui.popup_notification(id!(popup)).close(cx);
                }
                _ => {}
            }
```